### DISCUSSION TITLE
Zod for string validations

### DISCUSSION THREAD

**Discussion by ioucyf** [colinhacks/zod#1991] [Q&A] [answered]
Zod recommends checking validator.js for further string validation refinements in the documentation. 

I'm using Zod exclusively for runtime validations. So my questions is this:

Does Zod use validator.js for string validations such as uid, datetime, etc under the hood? 

If not, is it more efficient (or better) to use validator.js instead of Zod's for string validations in general?

I'm mostly interested in integrating string validations into my web client, and therefore trying to figure which would be best to include in the client in terms of size and performance.

**Comment by JacobWeisenburger** [comment, marked as answer]
> Does Zod use validator.js for string validations such as uid, datetime, etc under the hood?

Zod has no runtime dependencies. So it doesn't use validator.js under the hood.
https://github.com/colinhacks/zod/blob/a57fc2da3c48a5a2270e9684017d8bc056af8a40/package.json#L8

---

> If not, is it more efficient (or better) to use validator.js instead of Zod's for string validations in general?
>
> I'm mostly interested in integrating string validations into my web client, and therefore trying to figure which would be best to include in the client in terms of size and performance.

In terms of size. A simple search on npm shows that info.
- Zod: https://www.npmjs.com/package/zod
- Validator: https://www.npmjs.com/package/validator

In terms of performance. I have never used validator.js. So I don't know.

---

If you found my answer satisfactory, please consider supporting me. Even a small amount is greatly appreciated. Thanks friend! 🙏
https://github.com/sponsors/JacobWeisenburger
